sdk/monitor/azure-monitor-opentelemetry/azure/monitor/opentelemetry/_configure.py

```diff
@@ -92,7 +93,7 @@ def configure_azure_monitor(**kwargs) -> None:
 
 
 def _setup_tracing(configurations: Dict[str, ConfigurationValue]):
-    resource = configurations[RESOURCE_ARG] # type: ignore
+    resource =  cast(Resource, configurations[RESOURCE_ARG]) # type: ignore
```

Comment:
Why do we need this cast?

Reply:
Mypy complained. I am not sure why it waited until now. But casting the condigurations from Union to a clear type fixes it.
https://dev.azure.com/azure-sdk/public/_build/results?buildId=3413933&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=e3f8f308-6e5f-58aa-3a2a-fe425958de32&l=360

Reply:
Changed